dataplane/work_runner.h

```diff
@@ -0,0 +1,79 @@
+#pragma once
+#include <chrono>
+#include <iostream>
+#include <thread>
+#include <tuple>
+
+namespace dpdk
+{
+
+namespace internal
+{
+template<typename F>
+void CallIteration(F* f)
+{
+	f->Iteration();
+}
+
+template<typename F>
+void CallIteration(F f)
+{
+	f.Iteration();
+}
+
+} // namespace internal
+
+template<typename... Workloads>
+class WorkRunner
+{
+	std::tuple<Workloads...> m_workers;
+	bool m_running = true;
+	template<int I = 0>
+	void Iteration()
+	{
+		if constexpr (I != sizeof...(Workloads))
+		{
+			internal::CallIteration(std::get<I>(m_workers));
+			Iteration<I + 1>();
+		}
+	}
+
+public:
+	void Run()
+	{
+		while (m_running)
+		{
+			Iteration();
+		}
+	}
+	WorkRunner(Workloads... loads) :
+	        m_workers{loads...}
+	{
+	}
+};
+
+template<typename... Workloads>
+WorkRunner<Workloads...> MakeWorkRunner(Workloads&&... loads)
+{
+	return WorkRunner<Workloads...>{std::forward(loads)...};
+}
+
```

Comment:
I don't think this will compile. Using `std::forward` without specifying the template type seems incorrect. How is it supposed to know whether to use the copy or move constructors? `MakeWorkRunner` is an unused function, which might be why we haven't encountered compilation errors yet.